stan/math/rev/fun/simplex_constrain.hpp

```diff
@@ -93,9 +25,48 @@ class simplex_constrain_op {
  * @param y Free vector input of dimensionality K - 1
  * @return Simplex of dimensionality K
  */
-inline Eigen::Matrix<var, Eigen::Dynamic, 1> simplex_constrain(
-    const Eigen::Matrix<var, Eigen::Dynamic, 1>& y) {
-  return adj_jac_apply<internal::simplex_constrain_op>(y);
+template <typename T, require_eigen_col_vector_vt<is_var, T>* = nullptr>
+inline auto simplex_constrain(const T& y) {
+  using ret_type = plain_type_t<T>;
+
+  size_t N = y.size();
+
```

Comment:
WIth a function like this, I'd rather add:

```
if(N == 0) {
  return ret_type(y);
}
```

right here than have all the other code working with size zero stuff (it avoids at least one malloc too).

Reply:
Done

Reply:
Nevermind. For a zero length input we expect a length 1 output.